agent/local_test.go

```diff
@@ -1533,6 +1533,26 @@ func TestAgent_checkCriticalTime(t *testing.T) {
 	}
 }
 
+func TestAgent_AddCheckFailure(t *testing.T) {
+	t.Parallel()
+	cfg := TestConfig()
+	l := NewLocalState(cfg, nil)
+
+	// Add a check for a service that does not exist and verify that it fails
+	checkID := types.CheckID("redis:1")
+	chk := &structs.HealthCheck{
+		Node:      "node",
+		CheckID:   checkID,
+		Name:      "redis:1",
+		ServiceID: "redis",
+		Status:    api.HealthPassing,
+	}
+	if err := l.AddCheck(chk, ""); err == nil {
```

Comment:
@preetapan it would be good to look inside the error string and make sure you got the new error. It's easy to have some other error get caught with a "not `nil`" check, so that'll be a little more robust.